doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,451 @@
+    Routers need to be aware of the supported policy indices and the corresponding MPLS config they need to apply to packets to
+    enforce the policy in the internal network.
+    Routers periodically fetch this map from the control service.
+    A hash of the MPLS map is maintained, such that routers only have to update if their hash differs from the one at the control service.
+- RemotePolicyCache
+    When a local policy is queried at a remote AS, the resulting policy description is cached at the requesting AS' FABRID Manager,
+    such that subsequent requests can be served from cache.
+
+
+PCB dissemination
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The IndexIdentifierMap and SupportedIndicesMap are included in a (unsigned) detachable extension in the PCBs for an AS.
+Hashes of these maps are maintained in a Signed AS Entry, such that the authenticity of these maps can be verified.
+The detachable extension can also be present in the PCB, i.e. it does not have to be detached in all cases, e.g. if there are only very few policies.
+If the maps are detached, they can be fetched from the control service of that AS and the received maps can be verified with the hashes.
+To ensure a consistent hash calculation, the key entries of these maps have to be sorted, such that they are accessed in a consistent order.
+
+Exposing policies to the end hosts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The path combinator finds the most recent FABRID map per AS among the received segments and subsequently uses this map to find the FABRID
+policies that are available for each interface pair of hops.
+This results in a set of PolicyIdentifiers per hop, which can then be used by the application, such as with the usage of a
+specific 'sequence' parameter which incorporates the policies.
+Once the application has decided which policies to use, it can craft a FABRID HBH extension and include this as an option when sending
+the packet.
+
+DRKey
+^^^^^^
+
+FABRID uses DRKey for computing the Encrypted Policy Indices, the FABRID Hop Validation Fields and the Path Validator.
+The routers use the fast key derivation side, whereas the endhosts will use the slow side.
+
+Configuration
+--------------
+
+Control service
+^^^^^^^^^^^^^^^^^^
+
+To be able to use DRKey, one has to configure the control service setting "drkey.level1_db" and "drkey.secret_value_db".
+Additionally, since the border routers will fetch the secret value from the control service, the control service also has to
+add the internal IP address of all border routers of the local AS to the DRKey delegation list for FABRID.
+
+This could look like this::
+
+    [drkey.level1_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-level1.db"
+
+    [drkey.secret_value_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-secret.db"
+
+    [drkey.delegation]
+    FABRID = [ "fd00:f00d:cafe::7f00:11", "fd00:f00d:cafe::7f00:12", "fd00:f00d:cafe::7f00:13"]
+
+The FABRID policies are configured in the control service.
```

Comment:
@jeltevanbommel

Reply:
See below

Reply:
```suggestion

Configuring FABRID Policies
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FABRID policies are configured in the control service using YAML files. A YAML configuration contains the information necessary to create entries in the SupportedIndicesMap, IndexIdentifierMap, IdentifierDescriptionMap (for local policies) and the MPLSMaps. Concretely the YAML file should contain the following entries:

* ``local`` (bool): Indicating whether the policy is a local policy (true) or global policy (false).
* ``local_description`` (string): The description that is fetched by remote AS'es for this specific policy. Only used for local policies, as for global policies this is stored in a global datastore. Required when ``local`` is true, ignored otherwise.
* ``local_identifier`` (integer): The integer identifier that this policy is known by. Required when ``local`` is true, ignored otherwise.
* ``global_identifier`` (integer): The integer identifier of the policy from the global datastore that this configured policy should implement. Required when ``local`` is false, ignored otherwise.
* ``connections`` (list of ConnectionPoints): The connection points on which this policy applies.


Connection Points
'''''''''''''''''''
A connection point in the YAML file is defined by providing the ingress and egress of the connection as well as the integer MPLS label that should be applied to enforce the policy on that connection. The egress can be either an interface, an IPv4/IPv6 prefix or a wildcard. The ingress of a connection point is limited to a wildcard or interface.The rationale behind this is that an IPv4/IPv6 ingress would indicate a packet coming from an endhost. Upon arrival at the border router, the packet would have already traversed the internal network and the border router would not be able to enforce a policy (e.g. by applying an MPLS label).

Example of a list of connection points:

::

    - ingress:
          type: interface
          interface: 2
      egress:
          type: interface
          interface: 1
      mpls_label: 1    
    - ingress:
          type: interface
          interface: 2
      egress:
          type: ipv4
          ip: 192.168.5.1
          prefix: 24
      mpls_label: 55
    - ingress:
          type: wildcard
      egress:
          type: interface
          interface: 2
      mpls_label: 3


```